R/module_nested_tabs.R

```diff
@@ -188,12 +188,40 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, reporter) {
   module_reactive <- reactive({
     modules$server_args <- teal.transform::resolve_delayed(modules$server_args, datasets)
     is_module_server <- isTRUE("id" %in% names(formals(modules$server)))
-    args <- c(list(id = id, datasets = datasets), modules$server_args)
-
+    args <- c(list(id = id), modules$server_args)
     if (is_reporter_used(modules)) {
       args <- c(args, list(reporter = reporter))
     }
 
+    is_datasets_used <- isTRUE("datasets" %in% names(formals(modules$server)))
+    if (is_datasets_used) {
+      args <- c(args, datasets = datasets)
+    }
+
+    is_data_used <- isTRUE("data" %in% names(formals(modules$server)))
+    if (is_data_used) {
+      datanames <- if (identical("all", modules$filter)) datasets$datanames() else modules$filter
+
+      # list of reactive filtered data
+      data <- sapply(
+        datanames,
+        simplify = FALSE,
+        function(x) {
+          #todo: need to include metadata, keys, datalabel
+          reactive(datasets$get_data(x, filtered = TRUE))
+        }
+      )
+      #names(data) <- paste0(names(data), "_FILTERED") # these datasets are filtered
+
+      # code from previous stages
+      attr(data, "code") <- reactive(get_datasets_code(datanames, datasets))
+
+      # join_keys
+      attr(data, "join_keys") <- reactive(datasets$get_join_keys()[datanames])
+
+      args <- c(args, data = list(data))
```

Comment:
Something doesn't look quite right here - is `data` a list from the `sapply` which you are then adding the attributes onto but then `args` is setting `data = list(data)`

Reply:
This is a consequence of how `c` and `append` works with lists. `c` rather appends elements of the list instead of adding a whole list. If I didn't do it, `args` would contain `adsl = <reactive>`, `adtte = <reactive>` etc. instead of `data`. 

![image](https://user-images.githubusercontent.com/6959016/175907120-5849522c-354b-4a53-a2eb-1f98c74bb521.png)

Reply:
fair enough - good old R...